e2e/windows/e2e-starter.sh

```diff
@@ -8,7 +8,7 @@ log "Starting e2e tests"
 
 # Create a resource group for the cluster
 log "Creating resource group"
-RESOURCE_GROUP_NAME="$RESOURCE_GROUP_NAME"-"$WINDOWS_E2E_IMAGE"-v2
+RESOURCE_GROUP_NAME="$RESOURCE_GROUP_NAME-$WINDOWS_E2E_IMAGE-$K8S_VERSION"
```

Comment:
Per the offline discussion, we should not set the default k8s version in pipeline variables so it should be empty by default.

Reply:
Made changes.

k8s version in pipeline variables defaults to empty "".

If windows package url is set, `KUBERNETES_VERSION` must be set otherwise it returns error. This is to avoid using default version like `1.25.6` if intending to test a windows package version $e.g.$ `1.26.0`. Then we check if the package version is the same as the k8s version in pipeline variable. If it differs we return in invalid scenario.

If windows package url is not set, `KUBERNETES_VERSION` will be the windows package version. And if it's empty we populate it with the default version returned by the command `1.25.6` now as of Apr 20, 2023.